mac/build.sh

```diff
@@ -445,43 +242,88 @@ if $PREPRELEASE || $NOTARIZE; then
     echo
     builder_heading "Attempting to staple notarization to Keyman.app"
     xcrun stapler staple "$TARGET_APP_PATH" || builder_die "stapler failed"
+
+    DID_NOTARIZE=true
   fi
-fi
+}
 
-### Deploy as requested ###
-
-if $LOCALDEPLOY ; then
-    builder_heading "Attempting local deployment with command:"
-    KM4MIM_APP_BASE_PATH="$KM4MIM_BASE_PATH/build/$CONFIG"
-    displayInfo "$KM4MIM_BASE_PATH/localdeploy.sh \"$KM4MIM_APP_BASE_PATH\""
-    eval "$KM4MIM_BASE_PATH/localdeploy.sh" "$KM4MIM_APP_BASE_PATH"
-    if [ $? == 0 ]; then
-        displayInfo "Local deployment succeeded!" ""
-    else
-        builder_die "Local deployment failed!"
-    fi
-elif $PREPRELEASE ; then
-    builder_heading "Preparing files for release deployment..."
-    # Create the disk image
-    pushd setup
-    eval "./build.sh"
-    popd
-
-    eval "$KM4MIM_BASE_PATH/make-km-dmg.sh" $QUIET_FLAG
-    if [ $? == 0 ]; then
-        displayInfo "Creating disk image succeeded!" ""
-    else
-        builder_die "Creating disk image failed!"
-    fi
+do_sentry() {
+  # Upload symbols
+  builder_heading "Uploading symbols to sentry.keyman.com"
+
+  if which sentry-cli >/dev/null; then
+    pushd "$KM4MIM_BASE_PATH" > /dev/null
+    sentry-cli upload-dif "build/$CONFIG"
+
+    popd > /dev/null
+  else
+    builder_die "Error: sentry-cli not installed, download from https://github.com/getsentry/sentry-cli/releases"
+  fi
+
+  builder_finish_action success publish
+}
 
-    # Create download info
-    eval "$KM4MIM_BASE_PATH/write-download_info.sh"
-    if [ $? == 0 ]; then
-        displayInfo "Writing download_info file succeeded!" ""
-    else
-        builder_die "Writing download_info file failed!"
+do_install() {
+  if ! builder_has_option --quick; then
+    do_notarize
```

Comment:
Note:  `./build.sh install --debug` will fail if not `--quick`.  See `do_notarize`, which fails when code signing is disabled (currently, "new" line 215)... which is part of debug builds.

The conversion to the format does mean we can no longer do code-signed `--debug` builds.  Push come to shove, we could add a builder-option to force it, though.

Perhaps it would be better to flip the option and/or condition the install-type based on `--debug`?  Talking with @sgschantz, he mentioned doing most local builds the equivalent of `--debug` and `--quick`.

If based on `--debug`, I'm not sure how to best give instructions to resolve the error regarding "Disable SecAssessment".  It existed before (see "old" line 270), and I don't want to break something where I'm still a bit unclear.

----

I currently believe that dropping `--quick` and basing install-action based notarization on `--debug` is likely best... or at least, most convenient.  That's just my position though, and I can see good arguments being made to the contrary... so between that note and my lack of clarity on the error/warning, I maintained the complexity of the original script pending further feedback.

Reply:
Let's keep with the original design for those parameters for now and revisit once this is ready.